chroma-manager/tests/integration/installation_and_upgrade/test_update_with_yum.py

```diff
@@ -58,7 +58,8 @@ def test_no_retired_repos(self):
     # something we can run to clear the storage targets since this
     # test class doesn't use setUp()
     def test_clean_linux_devices(self):
-        self.cleanup_linux_devices(self.TEST_SERVERS)
+        self.cleanup_linux_devices([s for s in self.TEST_SERVERS
```

Comment:
why can't we use config_servers here?

Reply:
Because `self.TEST_SERVERS` is not always all of the servers in the _config_.  It's definable per test [class], which in and of itself is a bug with how `tearDown()` and `setUp()` currently work.

The bug is that it's actually `setUp()` (i.e. the start of a new test) that does all of this device reset/clearing and it does it across the `TEST_SERVERS` defined for that test.  But `TEST_SERVERS` might be different for this new test than it was for the previous test so when `setUp()` cleans those devices, it's ignoring the devices that were in the previous test but that are not in this new test.  But not cleaning those devices can have ripple effects in this new test.

I believe the fix is to move all of this device resetting/cleaning to `tearDown()` so that all of the devices that were used in a test are cleaned up at the end of that test while the configuration still matches what was in use for that test.  But that is a bigger can of worms than this patch will tackle.

Reply:
in this case it seems that we are duplicating `self.config_servers` as `self.TEST_SERVERS` here refers to that defined in *api_testcase_with_test_reset.py*

Reply:
I'm definitely not saying there is not duplication and confusion among these in the test framework in general which needs to be sorted out.  This just isn't the ticket for it.